src/Core/src/Platform/Android/MauiTextView.cs

```diff
@@ -20,21 +21,16 @@ internal override void OnLayoutFormatted(bool changed, int l, int t, int r, int
 		}
 		protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
 		{
-			if (MeasureSpec.GetMode(widthMeasureSpec) == MeasureSpecMode.AtMost && Layout is not null)
+			if (MeasureSpec.GetMode(widthMeasureSpec) == MeasureSpecMode.AtMost && !string.IsNullOrEmpty(Text))
 			{
-				// Ensure the Layout is valid and measured before reading LineCount or GetLineWidth(i) to avoid unnecessary calculations.
-				if (Layout.Width > 0)
-				{
-					// Calculate the total width needed based on text content plus padding
-					int contentWidth = (int)Math.Ceiling(GetMaxLineWidth(Layout));
-					int totalPadding = CompoundPaddingLeft + CompoundPaddingRight;
-					int requiredWidth = contentWidth + totalPadding;
-
-					// Constrain to maximum available width from original spec
-					int availableWidth = MeasureSpec.GetSize(widthMeasureSpec);
-					int desiredWidth = Math.Min(requiredWidth, availableWidth);
-					widthMeasureSpec = MeasureSpec.MakeMeasureSpec(desiredWidth, MeasureSpecMode.AtMost);
-				}
+				int availableWidth = MeasureSpec.GetSize(widthMeasureSpec);
+				int totalPadding = CompoundPaddingLeft + CompoundPaddingRight;
+				var layout = TextLayoutUtils.CreateLayout(Text, Paint, availableWidth - totalPadding, Android.Text.Layout.Alignment.AlignNormal);
+				int contentWidth = (int)Math.Ceiling(GetMaxLineWidth(layout));
+				// Calculate the required width based on the content and padding
+				int requiredWidth = contentWidth + totalPadding;
+				int desiredWidth = Math.Min(requiredWidth, availableWidth);
+				widthMeasureSpec = MeasureSpec.MakeMeasureSpec(desiredWidth, MeasureSpecMode.AtMost);
```

Comment:
Assumes that desiredWidth is always valid. If padding is larger than available width, does this result in unexpected behavior?

Reply:
I tested with a higher padding value, and it behaves the same as before the fix. 
However, I have restricted the creation of a new layout if the padding exceeds the available width, as base.Measure can handle this scenario. @jsuarezruiz